corehq/apps/enterprise/views.py

```diff
@@ -323,3 +329,65 @@ def refresh_item(self, item_id):
 
     def post(self, *args, **kwargs):
         return self.paginate_crud_response
+
+
+@require_can_edit_or_view_web_users
+@require_superuser
+def enterprise_permissions(request, domain):
+    account = BillingAccount.get_account_by_domain(domain)
+    all_domains = set(account.get_domains())
+    ignored_domains = set(account.permissions_ignore_domains)
+    controlled_domains = all_domains - ignored_domains - set([account.permissions_source_domain])
+
+    context = {
+        'domain': domain,
+        'all_domains': sorted(all_domains),
+        'source_domain': account.permissions_source_domain,
+        'ignored_domains': sorted(list(ignored_domains)),
+        'controlled_domains': sorted(list(controlled_domains)),
+        'current_page': {
+            'page_name': _('Enterprise Permissions'),
+            'title': _('Enterprise Permissions'),
+        }
+    }
+    return render(request, "enterprise/enterprise_permissions.html", context)
+
+
+@require_superuser
+@require_POST
+def toggle_enterprise_permission(request, domain, target_domain):
+    account = BillingAccount.get_account_by_domain(domain)
+    domains = account.get_domains()
+
+    redirect = reverse("enterprise_permissions", args=[domain])
+    if target_domain not in domains:
+        messages.error(request, _("Could not update permissions."))
+        return HttpResponseRedirect(redirect)
+    if target_domain in account.permissions_ignore_domains:
+        account.permissions_ignore_domains.remove(target_domain)
+    else:
+        account.permissions_ignore_domains.append(target_domain)
+    account.save()
+    messages.success(request, _('Permissions saved.'))
+    return HttpResponseRedirect(redirect)
+
+
+@require_superuser
+@require_POST
+def update_enterprise_permissions_source_domain(request, domain):
+    source_domain = request.POST.get('source_domain')
+    redirect = reverse("enterprise_permissions", args=[domain])
+
+    account = BillingAccount.get_account_by_domain(domain)
+    if source_domain and source_domain not in account.get_domains():
+        messages.error(request, _("Please select a valid domain."))
+        return HttpResponseRedirect(redirect)
+
+    if source_domain:
+        account.permissions_source_domain = source_domain
+    else:
+        account.permissions_source_domain = None
+        account.permissions_ignore_domains = []
+    account.save()
+    messages.success(request, _('Permissions saved.'))
```

Comment:
nit: message could be a bit more explicit

Reply:
part of 351136851cf787a2834805954a32ec493aaf5433